sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -251,6 +252,7 @@ namespace Azure { namespace Core { namespace Http {
     void AddQueryParameter(std::string const& name, std::string const& value);
     void AddHeader(std::string const& name, std::string const& value);
     void StartRetry(); // only called by retry policy
+    void SetDownloadViaStream(bool value) { this->m_isDownloadViaStream = value; }
```

Comment:
Can this be a ctor parameter instead? This allows it to be changeable multiple times after construction which can lead to unpredictable behavior.

Reply:
yes, changed!